incydr/cli/cmds/cases.py

```diff
@@ -117,10 +256,212 @@ def bulk_update(ctx: Context, csv: Path):
         console.print(f"[red]Error:[/red] {err.response.text}")
 
 
-# TODO: convert to click commands
+@cases.command(cls=IncydrCommand)
+@click.argument("case_number")
+@path_option
+@click.pass_context
+def download_summary(ctx: Context, case_number: int, path: str):
```

Comment:
I just had a thought - I wonder if this should just be a single `download` command that allows people to specify which files they want to download with flags (`--summary`, `--source-file`, `--events`, `--all`) and then we just adjust which api we call accordingly?

Reply:
Yeah, that would work, so:

- `incydr cases download 21 --summary` would call `download_summary_pdf()`
- `incydr cases download 21 --file-events` would call `download_file_events_csv()`
- `incydr cases download 21 --source-files` would call `download_full_case_zip()` with just `include_files=True`
- anytime more than one option is provided, `download_full_case_zip()` would also be called.

Reply:
Thats almost correct I think it would be more like

- `incydr cases download 21 --summary` would call `download_summary_pdf()`
- `incydr cases download 21 --file-events` would call `download_file_events_csv()`
- `incydr cases download 21 --source-files` would call `download_file_for_event`
- anytime more than one option is provided, `download_full_case_zip()` would be called with the appropriate include params
- `incydr cases download 21 --all` would call `download_full_case_zip()` with all files included